iml-manager-cli/src/display_utils.rs

```diff
@@ -88,3 +89,54 @@ where
 
     table
 }
+
+pub trait IntoTable {
+    fn into_table(self) -> Table;
+}
+
+impl IntoTable for Vec<Host> {
+    fn into_table(self) -> Table {
+        generate_table(
+            &["Id", "FQDN", "State", "Nids"],
+            self.into_iter().map(|h| {
+                vec![
+                    h.id.to_string(),
+                    h.fqdn,
+                    h.state,
+                    h.nids.unwrap_or_default().join(" "),
+                ]
+            }),
+        )
+    }
+}
+
+pub trait IntoDisplayType {
+    fn into_display_type(self, display_type: DisplayType) -> String;
+}
+
+impl<T: IntoTable + serde::Serialize> IntoDisplayType for T {
+    fn into_display_type(self, display_type: DisplayType) -> String {
+        match display_type {
+            DisplayType::Json => {
+                serde_json::to_string_pretty(&self).expect("Cannot serialize item to JSON")
+            }
+            DisplayType::Yaml => {
+                serde_yaml::to_string(&self).expect("Cannot serialize item to YAML")
+            }
+            DisplayType::Tabular => self.into_table().to_string(),
+        }
+    }
+}
+
+#[derive(StructOpt, Debug)]
+pub enum DisplayType {
+    Json,
+    Yaml,
```

Comment:
Nit: technically, YAML is a superset of JSON, so having just JSON would be enough.

Reply:
Depends on who you ask :)